docs/bookstore/page-1/pg1-review-svc-1.md

```diff
@@ -0,0 +1,379 @@
+    'ce-specversion': receivedEvent.specversion,
+    'ce-type': receivedEvent.type,
+    'ce-source': receivedEvent.source,
+    'ce-id': receivedEvent.id,
+  },
+  body: JSON.stringify(receivedEvent.data),
+});
+```
+
+![Image](images/image21.png)
+
+**Exploring Other Functions**
+
+The Node.js server contains other functions that follow a similar pattern, with detailed comments explaining their functionality:
+
+- `/insert`: Receives CloudEvents and inserts the payload into the PostgreSQL database.
+- `/comment`: Creates a WebSocket connection with the frontend to transport comments from the database.
+
+### **Step 2: Create Broker**
+
+![Image](images/image17.png)
+
+The Broker acts as a router in your event-driven application, receiving events and routing them to the correct destination.
+
+
+
+- 1: Create a new YAML file named `node-server/config/200-broker.yaml` and add the following content:
+
+???+ abstract "_node-server/config/200-broker.yaml_"
+    ```yaml
+    apiVersion: eventing.knative.dev/v1
+    kind: Broker
+    metadata:
+    name: bookstore-broker
+    ```
+
+- 2: Apply the YAML file:
+
+```bash
+kubectl apply -f node-server/config/200-broker.yaml
+```
+
+![Image](images/image16.png)
+
+Alternatively, use the [Knative CLI `kn`](https://knative.dev/docs/client/#kn) to create the broker:
+
+```bash
+kn broker create bookstore-broker
+```
+
+![Image](images/image19.png)
+
+???+ success "Verify"
+
+    ```bash
+    kubectl get brokers
+    ```
+
+    ![Image](images/image14.png)
+
+
+
+???+ bug "Troubleshooting"
+
+    If there are issues, use the following command to diagnose:
+
+    ```bash
+    kubectl describe broker bookstore-broker
+    ```
+
+### **Step 3: Create SinkBinding between Node.js server and broker**
+
+![Image](images/image15.png)
+
+Hardcoding URLs to connect with Kubernetes services in your application can be limiting and inflexible. SinkBinding dynamically injects the URL of the Kubernetes service into your application.
+
+Learn more about SinkBinding [here](https://knative.dev/docs/eventing/custom-event-source/sinkbinding/) and the [spec schema](https://knative.dev/docs/eventing/custom-event-source/sinkbinding/reference/)!
+
+**Create a SinkBinding:**
+
+- 1: Create a new YAML file named `300-sinkbinding.yaml` in the `node-server/config` folder and add the following content:
+
+???+ abstract "_node-server/config/300-sinkbinding.yaml_"
+    ```yaml
+    apiVersion: sources.knative.dev/v1
+    kind: SinkBinding # SinkBinding is a Knative Eventing custom resource
+    metadata:
+    name: node-sinkbinding
+    namespace: default
+    spec:
+    subject: # The subject is where we hope to inject the Sink URI
+        apiVersion: apps/v1
+        kind: Deployment
+        selector:
+        matchLabels:
+            app: node-server
+    sink: # The Sink is the URI of the Broker
+        ref:
+        apiVersion: eventing.knative.dev/v1
+        kind: Broker
+        name: bookstore-broker
+    ```
```

Comment:
The indentation of multiple fields is wrong here (spec, matchLabels, ref, etc)

Reply:
[sinkbinding.yaml.txt](https://github.com/user-attachments/files/15789234/sinkbinding.yaml.txt)